File: 1001-1500/1343_numOfSubarrays.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        summ = 0
        count = 0
        i = 0
        while i < len(arr):
            if i < k:
                summ += arr[i]
                i += 1
                if i == len(arr):
                    if summ / k >= threshold:
                        count += 1
                continue
            if summ / k >= threshold:
                count += 1
            summ += arr[i] - arr[i-k]
            i += 1
            if i == len(arr):
                if summ / k >= threshold:
                    count += 1
        return count

# Alternative solution

class Solution:
    def numOfSubarrays(self, arr, k, threshold):
        windowSum = sum(arr[:k])
        count = 1 if windowSum / k >= threshold else 0
        for i in range(len(arr) - k):
            windowSum += arr[i + k] - arr[i]
            if windowSum / k >= threshold:
                count += 1
        return count

File: 1001-1500/1343_numOfSubarrays.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int count = 0;
        double windowSum = 0;
        for (int i=0; i<k; i++) windowSum += arr[i];
        double average = windowSum / k;
        if (average >= threshold) count++;
        for (int i=k; i<arr.length; i++) {
            windowSum += arr[i] - arr[i-k];
            average = windowSum / k;
            if (average >= threshold) count++;
        }
        return count;
    }
}
